xdsl/backend/riscv/lowering/riscv_arith_lowering.py

```diff
@@ -78,34 +93,78 @@ def match_and_rewrite(
         rewriter.replace_matched_op([], [op.input])
```

Comment:
```python
class LowerArithIndexCast(RewritePattern):
    @op_type_rewrite_pattern
    def match_and_rewrite(
        self, op: arith.IndexCastOp, rewriter: PatternRewriter
    ) -> None:
        rewriter.replace_matched_op(
            [UnrealizedConversionCastOp.get(op.operands, (op.result.type,))]
        )
``` 
I think we also need a cast here otherwise the partial lowering might be invalid in some cases.